File: Notifier/Notifier.java
public class Notifier {
    private ApiService apiService;

    //initialize api service
    public Notifier(ApiService apiService) {
        this.apiService = apiService;
    }

    //sends an in-app notification to a user based on userId, this will be done via firebase cloud messaging
    public void sendNotification(String userId, String message) {
        apiService.sendNotification(userId, message);
    }

}

//interface that defines notification methods
public interface ApiService {
    void sendNotification(String userId, String message);
}

File: Notifier/Notifier.swift
class Notifier {
    let apiService: ApiService

    // Constructor to initialize the ApiService
    init(apiService: ApiService) {
        self.apiService = apiService
    }

    //sneds an in app notification to the user based on userId via firebase cloud messaging
    func sendNotification(to userId: String, with message: String) {
        apiService.sendNotification(to: userId, with: message)
    }
}

//protocol that defines notification methods
protocol ApiService {
    func sendNotification(to userId: String, with message: String)
}
